contracts/Penalizer.sol

```diff
@@ -23,14 +25,36 @@ contract Penalizer is IPenalizer{
         transaction.value,
         transaction.data) = RLPReader.decodeTransaction(rawTransaction);
         return transaction;
+    }
+
+    mapping(bytes32 => uint) public commits;
 
+    event Commit(bytes32 indexed commitHash, address sender);
+
+    /**
+     * any sender can call "commit(keccak(encodedPenalizeFunction))", to make sure
+     * no-one can front-run it to claim this penalization
+     */
+    function commit(bytes32 msgDataHash, uint revealSalt) external override {
```

Comment:
No, don't reveal salt :)

Salt should be revealed in 'reveal' stage.

'hash' should include secret only known to the signer.